### DISCUSSION TITLE
How do I work with CheckButton?

### DISCUSSION THREAD

**Discussion by RagnarTheEverchosen** [fltk-rs/fltk-rs#1267] [Q&A] [answered]
Hello, this might be kind of a trivial solve but for the love of god I can't figure out how can I make the user toggle on/off the CheckButton. 

Is it something I have to enable or could it be by having the button in a wring frame?

**Comment by MoAlyousef** [comment, marked as answer]
Hello

The user can toggle/untoggle the button by just clicking it.
If the CheckButton isn't clickable, it might not be inside the border of whatever is containing it. Can't really judge without looking at any code.
You can also toggle a button programmatically if you want.
```rust
use fltk::{prelude::*, *};

fn main() {
    let app = app::App::default();
    let mut win = window::Window::default().with_size(400, 300);
    let mut check = button::CheckButton::default()
        .with_size(80, 30)
        .with_label("button")
        .center_of_parent();
    check.set_checked(true);
    win.end();
    win.show();

    app.run().unwrap();
}
```

On launch:

![image](https://user-images.githubusercontent.com/37966791/183065393-97f5630c-d201-4ca4-838a-09bc1e7ad8ce.png)

After clicking:

![image](https://user-images.githubusercontent.com/37966791/183065452-8d5fc65d-a8ec-4f08-ba87-9d38f5cbc3ea.png)
